include/aikido/control/BarrettFingerKinematicSimulationPositionCommandExecutor.hpp

```diff
@@ -8,67 +8,70 @@
 #include <future>
 #include <mutex>
 #include <condition_variable>
+#include <chrono>
 
 namespace aikido {
 namespace control {
 
-/// This executor mimics the behavior of BarretFinger. 
-/// It moves a finger to a desired point; it may stop early if 
+/// This executor mimics the behavior of BarretFinger.
+/// It moves a finger to a desired point; it may stop early if
 /// joint limit is reached or collision is detected.
 /// Only the proximal joint is actuated; the distal joint moves with mimic ratio.
 /// When collision is detected on the distal link, the finger stops.
 /// When collision is detected on the proximal link, the distal link moves
 /// until it reaches joint limit or until distal collision is detected.
-class BarrettFingerPositionCommandExecutor
+class BarrettFingerKinematicSimulationPositionCommandExecutor
 {
 public:
   /// Constructor.
-  /// \param _finger Finger to be controlled by this Executor.
-  /// \param _proximal Index of proximal dof
-  /// \param _distal Index of distal dof
-  /// \param _collisionDetector CollisionDetector for detecting self collision
-  ///        and collision with objects. 
-  /// \param _collisionOptions Default is (enableContact=false, binaryCheck=true,
-  ///        maxNumContacts = 1.) 
-  ///        See dart/collison/Option.h for more information 
-  BarrettFingerPositionCommandExecutor(
-    ::dart::dynamics::ChainPtr _finger, int _proximal, int _distal,
-    ::dart::collision::CollisionDetectorPtr _collisionDetector,
-    ::dart::collision::CollisionOption _collisionOptions
+  /// \param[in] finger Finger to be controlled by this Executor.
+  /// \param[in] proximal Index of proximal dof
+  /// \param[in] distal Index of distal dof
+  /// \param[in] collideWith CollisionGroup to check collision with fingers.
+  /// \param[in] collisionOptions Default is (enableContact=false, binaryCheck=true,
+  ///        maxNumContacts = 1.)
+  ///        See dart/collison/Option.h for more information
+  BarrettFingerKinematicSimulationPositionCommandExecutor(
+    ::dart::dynamics::ChainPtr finger, size_t proximal, size_t distal,
+    ::dart::collision::CollisionGroupPtr collideWith,
```

Comment:
Got it, thank you for the clarification! I've rewritten the logic to add the fourth case. :smiley: